File: ascending_trend.py
import sys
import statistics


if len(sys.argv) < 2:
    sys.exit('This program needs exactly one argument: "INT_STRING"')

try:
    int(sys.argv[1])
except ValueError as e:
    print(f'{e} Invalid INT_STRING')
    sys.exit(1)

int_string = sys.argv[1]
str_segments = ['' for i in range(len(int_string))]
segment_index = 0
str_segments[segment_index] += int_string[0]
counter = 1

for i in range(len(int_string)):
    if counter == len(int_string):
        break
    counter += 1
    if int(int_string[i+1]) > int(int_string[i]):
        str_segments[segment_index] += int_string[i+1]
    else:
        segment_index += 1
        str_segments[segment_index] += int_string[i+1]

longest = max(str_segments, key=len)
average = statistics.mean([int(i) for i in list(longest)])

print(
    f'Longest substring in numeric ascending order is: {longest} Average: '
    f'{average:.2f}'
    )

File: AscendingTrend.java
import java.util.*;

public class AscendingTrend {

    public static void main(String[] args) {
        
        if (args.length < 1) {
            System.out.println("This program needs exactly one argument: INT_STRING");
            System.exit(1);
        }
        String intString = args[0];
        ArrayList<String> strSegments = new ArrayList<String>();
        int segmentIndex = 0;
        
        for (int i = 0; i < intString.length(); i++) {
            strSegments.add("");
        }
        strSegments.set(segmentIndex, strSegments.get(segmentIndex) + intString.charAt(0));
        int counter = 1;

        for (int j = 0; j < intString.length(); j++) {
            if (counter == intString.length()) {
                break;
            }
            counter++;

            if ((int) intString.charAt(j) < (int) intString.charAt(j+1)) {
                strSegments.set(segmentIndex, strSegments.get(segmentIndex) + intString.charAt(j+1));
            } else {
                segmentIndex++;
                strSegments.set(segmentIndex, strSegments.get(segmentIndex) + intString.charAt(j+1));
            }
        }

        String longest = strSegments.get(0);

        for (String segment : strSegments) {
            if (segment.length() > longest.length()) {
                longest = segment;
            }
        }

        int strLength = longest.length();
        int sum = 0;

        for (int k = 0; k < strLength; k++) {
            sum += (int) longest.charAt(k);
        }

        float average = (float) sum / (float) strLength;

        System.out.println(String.format("Longest substring in numeric ascending order is: %s Average: %.2f", longest, average));


    }
}